src/main/java/com/scaleunlimited/flinkcrawler/warc/CreateWARCWritableFunction.java

```diff
@@ -0,0 +1,111 @@
+package com.scaleunlimited.flinkcrawler.warc;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.Collector;
+import org.apache.hadoop.io.NullWritable;
+
+import com.scaleunlimited.flinkcrawler.functions.BaseFlatMapFunction;
+import com.scaleunlimited.flinkcrawler.pojos.FetchResultUrl;
+import com.scaleunlimited.flinkcrawler.pojos.FetchStatus;
+
+@SuppressWarnings("serial")
+public class CreateWARCWritableFunction
+        extends BaseFlatMapFunction<FetchResultUrl, Tuple2<NullWritable, WARCWritable>> {
+
+    private static String SOFTWARE = "flink-crawler";
+
+    private SimpleDateFormat _dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+    private String _userAgentString;
+    private boolean _warcInfoEmitted;
```

Comment:
Perhaps _isWarcInfoEmitted (or _isWarcInfoAlreadyEmitted?) would be a better name?

Reply:
I'd be OK with that convention if you *always* used the getter and setter methods (even within the class itself). Otherwise, I'm reading code that's using a boolean without the identifier looking like a boolean.